src/spikeinterface/core/basesorting.py

```diff
@@ -424,6 +427,31 @@ def get_all_spike_trains(self, outputs="unit_id"):
             spikes.append((spike_times, spike_labels))
         return spikes
 
+    def precompute_spike_trains(self, from_spike_vector: bool = True):
```

Comment:
Since this is in core where numba is not automatically installed wouldn't it be safer to have `from_spike_vector` default to False, so that a user who has only installed core doesn't instantly hit an assert error?

Reply:
I added the parameter to be complete, but I actually don't know when it would be interesting to precompute spike trains not from the spike vector.
Can you think of a use case?

Reply:
Not off the top of my head. But I'll think about it :)